docs/_documentations/codewind-minikube-guickguide.md

```diff
@@ -0,0 +1,198 @@
+* Install Codewind 
+* Develop a microservice with Codewind on Minikube 
+
+## Overview
+Minikube is essentially Kubernetes but on your local computer. Minikube supports the latest Kubernetes release, so you can configure Minikube on any container runtime. With Minikube, you can start your computer then utilize all Kubernetes' capabilities. Use Minikube to explore and learn to develop with Codewind. This quick guide assists and instructs you to deploy Codewind and projects on Minikube. 
+
+* [Prerequisistes](##prerequisites)
+* [Installing Minikube](##installing-minikube)
+* [Installing Codewind](##installing-codewind)
+* [Developing a microservice](##developing-a-microservice)
+* [What you have learned](##what-you-have-learned)
+* [Next steps](##next-steps) 
+
+## Prerequisites
+* A Mac, Windows, or Linux host.
+* Install [Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/).
+* At least 8Gb RAM. Minikube needs at least 4Gb of RAM for this quick guide. 
+
+## Installing Minikube
+1. To run the needed commands, open your terminal. 
+2. To start Minikube on your machine, run, `minikube start --memory=4096`.
+    * Minikube creates a single-node Kubernetes cluster with 4GB of RAM available to it.
+    * If this command fails, refer to [Minikube installation docs](https://kubernetes.io/docs/tasks/tools/install-minikube) and ensure you have enough memory on your system. 
+3. To install [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) on Minikube, run, `minikube addons enable ingress`. 
+    * Codewind makes use of Ingress to expose its endpoints.
+4. To determine your Ingress IP, run `minikube ip`. 
+    * Example:
+     ```
+    $ minikube ip
+    192.168.64.46
+     ```
+     * Save this value for later.
+ 
+## Installing Codewind
+1. Choose your IDE and install the Codewind extension. 
+    * Follow the install instructions for [Eclipse](https://www.eclipse.org/codewind/mdt-eclipse-getting-started.html) or [VS Code](https://www.eclipse.org/codewind/mdt-vsc-getting-started.html). 
+2. Locate the `cwctl` command line interface (CLI). 
+    * On Mac and Linux, the cwctl is `~/.codewind/<version>/cwctl`. 
+    * On Windows, the cwctl is `C:\Users\(your user name)\.codewind\<version>\cwctl`. 
+3. To install cwctl on Mac and Linux, run:
+        ```
+        ./cwctl --insecure install remote \
+        --namespace codewind \
+        --kadminuser admin \
+        --kadminpass admin \
+        --krealm codewind \
+        --kclient codewind \
+        --kdevuser developer \
+        --kdevpass developer \
+        --ingress <ingress-ip>.nip.io
+        ```
+4. To install cwctl on Windows, run: `cwctl.exe --insecure install remote --namespace codewind --kadminuser admin --kadminpass admin --krealm codewind --kclient codewind --kdevuser developer --kdevpass developer --ingress <ingress-ip>.nip.io`     
+    * **Note:** The `ingress-ip` value is the value you found in Step 4 of `Preparing Minikube`.
```

Comment:
When I ran step 3 for installing remote in cwctl, it seems to be taking forever.

I am seeing this as the output every time:
```WARN[1650] Watch for Pod codewind-keycloak-ka5ft7nq ended. Timeout or connection error. 
INFO[1650] Waiting for pod: codewind-keycloak-ka5ft7nq  
INFO[1650] codewind-keycloak-ka5ft7nq, phase: Pending   
INFO[1650] codewind-keycloak-ka5ft7nq, phase: Pending ContainersNotReady  ```

I have tried to delete and restart minikube, and run cwctl stop-all to see if that helps, but it didn't.

Even though this is a quick guide, I don't seem to see any troubleshooting tips for this.

Reply:
@naveenkaratekid Can you try following the instructions for the Codewind operator and seeing if they work for you? Wondering if the quicklab should use the operator isntead

Reply:
@johnmcollier what instructions for the Codewind operator are you referring to?

Reply:
Do you mean the instructions laid out here in [eclipse/codewind-operator](https://github.com/eclipse/codewind-operator)?

Reply:
@johnmcollier I went through the steps in the codewind operator, and when I ran the install script with the ingress IP as the argument, this is what I got:

```
Naveens-MacBook-Pro:deploy naveenk@ibm.com$ ./install.sh operator -i 192.168.64.12.nip.io
############################
Codewind Operator install.sh
############################

########################
   Codewind Operator
########################

Ingress Domain: 192.168.64.12.nip.io
Target Openshift 311: false
Creating Codewind namespace:
Error from server (AlreadyExists): namespaces "codewind" already exists
Deploying Operator Service Account:
Error from server (AlreadyExists): error when creating "service_account.yaml": serviceaccounts "codewind-operator" already exists
Deploying Operator RBAC Roles:
Error from server (AlreadyExists): error when creating "role.yaml": roles.rbac.authorization.k8s.io "codewind-operator" already exists
Deploying Operator RBAC Role Bindings:
Error from server (AlreadyExists): error when creating "role_binding.yaml": rolebindings.rbac.authorization.k8s.io "codewind-operator" already exists
Deploying Operator Cluster Roles:
Error from server (AlreadyExists): error when creating "cluster_roles.yaml": clusterroles.rbac.authorization.k8s.io "codewind-operator-cluster" already exists
Deploying Codewind Cluster Role Bindings:
Error from server (AlreadyExists): error when creating "cluster_role_binding.yaml": clusterrolebindings.rbac.authorization.k8s.io "codewind-operator-clb" already exists

Installing Custom Resource Definitions (CRD):
Error from server (AlreadyExists): error when creating "codewind.eclipse.org_keycloaks_crd.yaml": customresourcedefinitions.apiextensions.k8s.io "keycloaks.codewind.eclipse.org" already exists
Error from server (AlreadyExists): error when creating "codewind.eclipse.org_codewinds_crd.yaml": customresourcedefinitions.apiextensions.k8s.io "codewinds.codewind.eclipse.org" already exists
Creating Codewind configmap:
Error from server (AlreadyExists): error when creating "custom-codewind-configmap.yaml": configmaps "codewind-operator" already exists
Deploying Codewind operator:
Error from server (AlreadyExists): error when creating "operator.yaml": deployments.apps "codewind-operator" already exists
Requesting a new Keycloak service
Error from server (AlreadyExists): error when creating "codewind.eclipse.org_v1alpha1_keycloak_cr.yaml": keycloaks.codewind.eclipse.org "devex001" already exists
Reading Keycloak deployments
NAME       NAMESPACE   AGE   ACCESS
devex001   codewind    17s   
Waiting for keycloak (may take a few minutes Pending->ContainerCreating->Running)
keycloak 
```

Reply:
Moving this guide out to the next release.

Reply:
@naveenkaratekid Can you try the following?
```
$ minikube stop
$ minikube delete
$ minikube start --memory=4gb
```

Then reproduce the operator install steps

Reply:
@johnmcollier, I heard from Naveen that these instructions don't work either. Any other thoughts on what we can do to get Codewind installed on Minikube?